code/scripting/api/objs/ship.cpp

```diff
@@ -1717,13 +1717,13 @@ ADE_FUNC(getAltClassName, l_Ship, NULL, "Gets the alternate class name of the sh
 	if (shipp->alt_type_index < 0)
 		return ade_set_args(L, "s", "");
 
-	char temp[NAME_LENGTH];
+	char temp_altname[NAME_LENGTH];
 
-	*temp = 0;
-	mission_parse_lookup_alt_index(shipp->alt_type_index, temp);
+	*temp_altname = 0;
+	strcpy(temp_altname, mission_parse_lookup_alt_index(shipp->alt_type_index));
```

Comment:
```suggestion
	strcpy_s(temp_altname, mission_parse_lookup_alt_index(shipp->alt_type_index));
```
No need to initialize the char array since `strcpy` will do that anyway.